common/djangoapps/student/views/dashboard.py

```diff
@@ -781,6 +782,22 @@ def student_dashboard(request):
             enr for enr in course_enrollments if entitlement.enrollment_course_run.course_id != enr.course_id
         ]
 
+    course_upgrade_message = ""
+    if request.GET.get('transaction') and request.GET.get('data'):
+        data = request.GET.get('data')
+        decoded_data = data.decode('base64', 'strict')
+        transaction_course, transaction_user = decoded_data.split('#')
+        if transaction_course and transaction_user:
+            log.info(
+                "Approved Authorizenet transaction. Pending course status for Course %s and User %s",
+                transaction_user,
+                transaction_course,
+            )
+            if transaction_user == request.user.username:
+                transaction_course_id = CourseKey.from_string(transaction_course.replace(" ", "+"))
```

Comment:
Why we have to replace space with `+`? are not we urlencoding/decoding `transaction_course`?

Reply:
I have updated it.